musica/src/micm/micm.cpp

```diff
@@ -0,0 +1,74 @@
+#include <iostream>
+
+#include "../../include/micm/micm.hpp"  // TODO(jiwon) relative include path?
+#include <micm/configure/solver_config.hpp>
+
+
+MICM::MICM(const std::string& config_path)
+    : config_path_(config_path),
+      solver_(nullptr)
+    {
+        std::cout << "   * [C++] MICM constructor" << std::endl;
+    }
+
+MICM::~MICM()
+{
+    std::cout << "   * [C++] MICM Destructor" << std::endl;
+}
+
+int MICM::create_solver()
+{
+    std::cout << "   * [C++] Creating solver" << std::endl;
+    bool success = 1;    // TODO(jiwon): can we specifiy error type with int?
+
+    // read and parse the config
+    micm::SolverConfig solver_config;
+    micm::ConfigParseStatus status = solver_config.ReadAndParse(config_path_);
+    micm::SolverParameters solver_params = solver_config.GetSolverParams();
+
+    if (status == micm::ConfigParseStatus::Success)
+    {
+        auto params = micm::RosenbrockSolverParameters::three_stage_rosenbrock_parameters(NUM_GRID_CELLS);
+        params.reorder_state_ = false;
+        solver_ = new VectorRosenbrockSolver{solver_params.system_,
+                                            solver_params.processes_,
+                                            params}; 
+    }
+    else
+    {
+        std::cout << "   * [C++] Failed creating solver" << std::endl;
+        success = 0; 
+    }
+
+    return success;
+}
+
+void MICM::delete_solver() const
```

Comment:
Please make this a destructor so that in `delete_micm` you can just call `delete micm` and not have to care about calling this function

Reply:
Yeah, I updated the destructor.